spec/consistency_spec.rb

```diff
@@ -39,6 +39,29 @@
 
         specify { expect(subject.signature(array)).to eq md5 }
       end
+
+      context 'given stringified hash' do
+        let(:stringified_hash) {"{ a: { b: 'b', c: 'c' }, d: 'd' }"}
+        let(:md5) { 'f5327912eeb41996b5aded6550d11187' }
+
+        specify { expect(subject.signature(stringified_hash)).to eq md5 }
+      end
+
+      context 'given a string' do
+        let(:string) {"do shash'owania"}
+        let(:md5) { '48e380a165c417253512a904d6b5cf2b' }
+
+        specify { expect(subject.signature(string)).to eq md5 }
+      end
+
+      context 'given a number' do
+        let(:number) {1234567890}
+        # RUBY 2.5.0
+        # let(:md5) { 'cbabd6f8bfda13b76c0e28eb0a6f4ef1' }
```

Comment:
Does Ruby 2.5 generate a different hash?

Reply:
With Ruby 2.4:
```
Failed examples:

rspec ./spec/consistency_spec.rb:17 # Crimp.signature check MD5 consistent across versions given a Hash should eq "1dd744d51279187cc08cabe240f98be2"
rspec ./spec/consistency_spec.rb:33 # Crimp.signature check MD5 consistent across versions given an Array should eq "cd29980f258eef3faceca4f4da02ec65"
rspec ./spec/consistency_spec.rb:40 # Crimp.signature check MD5 consistent across versions given legacy Array should eq "4dc4e1161c9315db0bc43c0201b3ec05"
```

Reply:
this probably happens because in crimp.rb we stringify the object with:
```ruby 
 def self.to_string(obj)
    "#{obj}#{obj.class}"
  end
```

in ruby 2.1 an intger class is a `Fixnum` where in Ruby > 2.4 is a `Integer`

Reply:
Ruby 2.1
```ruby
2.class.ancestors
[Fixnum, Integer, Numeric, Comparable, Object, PP::ObjectMixin, Kernel, BasicObject]
```
Ruby 2.4.2
```ruby
2.class.ancestors
[Integer, Numeric, Comparable, Object, PP::ObjectMixin, Kernel, BasicObject]
```

Floats are not showing the same issues, as both 2.1 and 2.4 are returning:
```ruby
6.6.class.ancestors
[Float, Numeric, Comparable, Object, Kernel, BasicObject]
```

Reply:
Wondering if makes sense to have the class part of objects which are not Strings|Hashes|Arrays as this would make very hard to build a new Crimp library in a different language as would require some sort of mapping between object class names...

Reply:
I'll create a PR to fix the Fixnum issue so that Crimp can maintain backwards compatibility, but this calls for a review of the gem to make it more language/version agnostic.

Reply:
ah also there is still a possible issue with Bignum objects:

Ruby 2.1

```ruby
100000000000000000000000000000000000000000000000000000.class
=> Bignum
```

Ruby >= 2.4
```ruby
100000000000000000000000000000000000000000000000000000.class
=> Integer
```

But for now I'd try to ignore and review it with the new version